src/components/admin/cropper-modal/CropperModal.tsx

```diff
@@ -0,0 +1,221 @@
+import ReactCrop, { Crop, PixelCrop } from 'react-image-crop';
+import 'react-image-crop/dist/ReactCrop.css';
+import { useCallback, useEffect, useRef, useState } from 'react';
+import { Image, ImageValues } from '../../../types/common/image';
+import { Modal } from '../../common/modal/Modal';
+import { Button } from '../button/Button';
+import { COMMON_TEXT_ADMIN } from '../../../const/admin/common';
+import './CropperModal.scss';
+import { CROPPER_CONSTANTS } from '../../../const/admin/cropper';
+import { InlineLoader } from '../../common/inline-loader/InlineLoader';
+
+interface CropModalProps {
+    src: ImageValues | null;
+    onChange: (value: ImageValues) => void;
+    height: number;
+    width: number;
+    onCancel: () => void;
+    isOpen: boolean;
+}
+
+export const CropModal = ({ src, onChange, width, height, onCancel, isOpen }: CropModalProps) => {
+    const [crop, setCrop] = useState<Crop>();
+    const [displaySrc, setDisplaySrc] = useState<string | null>(null);
+    const [completedCrop, setCompletedCrop] = useState<PixelCrop>();
+    const imgRef = useRef<HTMLImageElement>(null);
+    const [rawImage, setRawImage] = useState<ImageValues | Image | null>(null);
+    const [naturalWidth, setNaturalWidth] = useState<number | null>(null);
+    const [renderedWidth, setRenderedWidth] = useState(0);
+    const aspectRatio = width / height;
+
+    useEffect(() => {
+        if (!src) {
+            setRawImage(null);
+            setDisplaySrc(null);
+            return;
+        }
+        setRawImage(src);
+        setDisplaySrc(`data:${src.mimeType};base64,${src.base64}`);
+
+        setCrop(undefined);
+        setCompletedCrop(undefined);
+    }, [src]);
```

Comment:
I believe it can be replaced by
```suggestion
    const displaySrc = useMemo(() => {
        if (!src) return null;
        return `data:${src.mimeType};base64,${src.base64}`;
    }, [src]);
```

Reply:
When new src is given from the outside, crop won't be cleared.